src/sentry/api/endpoints/organization_event_details.py

```diff
@@ -86,16 +87,43 @@ class OrganizationEventDetailsEndpoint(OrganizationEventsEndpointBase):
         "GET": ApiPublishStatus.PRIVATE,
     }
 
-    def get(self, request: Request, organization, project_slug, event_id) -> Response:
-        """event_id is validated by a regex in the URL"""
-        if not self.has_feature(organization, request):
-            return Response(status=404)
+    def convert_args(
```

Comment:
i'm curious why this neds its own convert args function -- what customization does this have that the parent class doesnt?

Reply:
The parent class that implements convert args is Organization Endpoint, which only converts the org. this will convert the project as well.

Reply:
could this just be a project endpoint then?

Reply:
hmm you maybe correct. Let  me make `OrganizationEventsEndpointBase`, inherit ProjectEndpoint and see if any tests fail.

Reply:
i don't think i can do that as there are a bunch of endpoints like `OrganizationEventsRelatedIssuesEndpoint` that also inherit `OrganizationEventsEndpointBase`. some of these endpoints don't have project_slug passed in, so i can't have the parent inherit project endpoint. I guess i could further refactor OrganizationEventsRelatedIssuesEndpoint but right now, i am trying to wrap up the id or slug stuff and this endpoint is semi blocking for it.

Reply:
got it

Reply:
Interesting. I don't think that's a good architecture in first place 😞 This endpoint should have been ProjectEventDetails if project is always required for it and always is singular. I'll create a github issue to take a look at the refactory.

Reply:
https://github.com/getsentry/sentry/issues/69686